pkg/registry/batch/job/strategy.go

```diff
@@ -122,20 +119,12 @@ func hasJobTrackingAnnotation(job *batch.Job) bool {
 	return ok
 }
 
-func dropJobTrackingAnnotation(job *batch.Job) {
-	delete(job.Annotations, batchv1.JobTrackingFinalizer)
-}
-
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (jobStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 	newJob := obj.(*batch.Job)
 	oldJob := old.(*batch.Job)
 	newJob.Status = oldJob.Status
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.JobTrackingWithFinalizers) && !hasJobTrackingAnnotation(oldJob) {
-		dropJobTrackingAnnotation(newJob)
-	}
```

Comment:
I thought we were going to prevent adding or removing on update with this:

```go
if !hasJobTrackingAnnotation(oldJob) {
  dropJobTrackingAnnotation(newJob)
} else {
  addJobTrackingAnnotation(newJob)
}
```